tiledb/sm/rest/test/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+#
+# tiledb/sm/rest/test/CMakeLists.txt
+#
+# The MIT License
+#
+# Copyright (c) 2025 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+include(unit_test)
+
+commence(unit_test rest_profile)
+    this_target_sources(main.cc unit_rest_profile.cc)
+    this_target_link_libraries(rest_profile tiledb_test_support_lib)
```

Comment:
Off-topic: it's not good that we have spent significant effort to componentize the codebase into lots of tiny object libraries in part to allow independently compiling and testing parts of the library (IIUC), but we still have not finished it, and as soon as you reference `tiledb_test_support_lib` you bring the whole `TILEDB_CORE_OBJECTS` and have to compile 269 files. 🥲

We should rethink the object libraries approach; with this PR we have 91 object libraries, a number of which I personally don't see the benefits.

Reply:
Perhaps we might consider migrating `TemporaryLocalDirectory` into `common` to mitigate some of the overhead. There will certainly be other complications involved in this work, but it's something we should discuss offline.